util/eventbus/debug.go

```diff
@@ -9,6 +9,147 @@ import (
 	"sync/atomic"
 )
 
+// A Debugger offers access to a bus's privileged introspection and
+// debugging facilities.
+//
+// The debugger's functionality is intended for humans and their tools
+// to examine and troubleshoot bus clients, and should not be used in
+// normal codepaths.
+//
+// In particular, the debugger provides access to information that is
+// deliberately withheld from bus clients to encourage more robust and
+// maintainable code - for example, the sender of an event, or the
+// event streams of other clients. Please don't use the debugger to
+// circumvent these restrictions for purposes other than debugging.
+type Debugger struct {
+	bus *Bus
+}
+
+// Clients returns a list of all clients attached to the bus.
+func (d *Debugger) Clients() []*Client {
+	return d.bus.listClients()
+}
+
+// PublishQueue returns the contents of the publish queue.
+//
+// The publish queue contains events that have been accepted by the
+// bus from Publish() calls, but have not yet been routed to relevant
+// subscribers.
+//
+// This queue is expected to be almost empty in normal operation. A
```

Comment:
The reference to "full" here makes me wonder if we should export descriptive methods like "queue capacity" (e.g., if I get a slice of 5 events here, is that "full" or just "busy"?)

Separate question, and not necessary for this PR.

Reply:
Hmm, good question. Generally I'm reluctant to expose the expected queue sizes, in the same vein as Fuchsia saying "there is some queuing, the amount isn't specified, don't worry about it. But for a debug facility, you're right that I can't really reason usefully about the slice I get back if it's non-zero. Hmm.

Let's tinker with this in a followup, unlike the other methods I haven't tried these out in anger yet, and doing so will probably guide towards a better state.

Reply:
Follow-up SGTM. I would also be fine with, say documenting `const MaxQueueSize = ...` and pointing to that. I don't feel like we would gain enough to be worth adding a flag here ("queue is full")